apps/designer/app/designer/features/style-panel/sections/spacing/spacing.tsx

```diff
@@ -111,61 +120,87 @@ const styles = {
 
 type TextFieldProps = {
   property: StyleProperty;
-  value: string | undefined;
+  value?: UnitValue;
   onEnter: (value: string) => void;
 };
 
+const initialValue: UnitValue = {
+  value: 0,
+  unit: "px",
+  type: "unit",
+};
+
+const slideUpAndFade = keyframes({
+  "0%": { opacity: 0, transform: "scale(0)" },
+  "100%": { opacity: 1, transform: "scale(1)" },
+});
+
 const TextField = ({ property, value, onEnter }: TextFieldProps) => {
-  const [currentValue, setCurrentValue] = useState<string>(value ?? "");
+  const [currentValue, setCurrentValue] = useState<UnitValue>(
+    value ?? initialValue
+  );
   const isFromCurrentBreakpoint = useIsFromCurrentBreakpoint(property);
 
   useEffect(() => {
-    setCurrentValue(value ?? "");
+    setCurrentValue(value ?? initialValue);
```

Comment:
Usually you don't need useEffect here [explanation](https://beta.reactjs.org/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes)

Reply:
good catch, didn't know this one